src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -94,6 +109,29 @@ public KNNQueryBuilder(String fieldName, float[] vector, int k, QueryBuilder fil
         this.k = k;
         this.filter = filter;
         this.ignoreUnmapped = false;
+        this.radius = 0.0f;
+    }
+
+    public KNNQueryBuilder(String fieldName, float[] vector, float radius, QueryBuilder filter) {
```

Comment:
may time to refactor to Builder pattern?

Reply:
Good point, let me check.

Reply:
after introducing radius, neither k or radius will be required

Reply:
But cant we switch to builder pattern still?

Reply:
Updated to use builder pattern in new code.